.github/workflows/if-nodejs-pr-testing.yml

```diff
@@ -72,3 +72,20 @@ jobs:
         name: Run release assets generation to make sure PR does not break it
         shell: bash
         run: npm run generate:assets --if-present
+
+  upload-coverage:
+    name: Upload Coverage to Codecov
+    runs-on: ubuntu-latest
+    needs: test-nodejs-pr
```

Comment:
Do we really need a different job for this? This can be added as a step also

Reply:
Code coverage will run three times—for macOS, Linux, and Windows which is not a good approach. So, I created a new job for it.

Reply:
Why tests shouldn't be run for all three OS? Previously in asyncapi website, the development setup was setup only for Linux, only because of some errors for Windows OS in the scripts. Hence, it's important to run tests in all three OS.